lib/apiservers/portlayer/restapi/handlers/storage_handlers.go

```diff
@@ -507,6 +515,142 @@ func (h *StorageHandlersImpl) VolumeJoin(params storage.VolumeJoinParams) middle
 	return storage.NewVolumeJoinOK().WithPayload(actualHandle.String())
 }
 
+func (h *StorageHandlersImpl) StatPath(params storage.StatPathParams) middleware.Responder {
+	// do offline container stat path, if fails do online
+	op := trace.NewOperation(context.Background(), fmt.Sprintf("StatPath(%s)", params.DeviceID))
+
+	fileStat, err := h.volumeCache.StatPath(op, *params.Store, params.DeviceID, params.TargetPath)
```

Comment:
I would prefer that we don't overload the cache with an operation like statpath, but I'll defer this to @jzt who will now own storage.  To me, a cache is just a lookup structure.